C++ error when building for android

### Description

Android build always fails, web works fine, but haven't tried iOS. I get the same error whether I build with expo or local EAS.
```
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /home/elfennani/Android/Sdk/cmake/3.22.1/bin/ninja \
        -C \
        /home/elfennani/projects/pickleball-lms/node_modules/react-native-unistyles/android/.cxx/Debug/1d194t4p/x86 \
        unistyles
    from /home/elfennani/projects/pickleball-lms/node_modules/react-native-unistyles/android
```


### Steps to Reproduce

1. create new expo project `yarn create expo-app`.
2. follow unistyle 3.0 installation instructions.
3. `yarn expo prebuild --clean -p android`
4. `yarn android`

### Snack or Repository Link (Optional)

https://github.com/elfennani/pickleball-lms

### Unistyles Version

3.0.0-beta.1

### React Native Version

0.76.3

### Platforms

Android

### Expo

Yes

I am having the same issue, here is the full error:

```
  FAILED: ../../../../build/intermediates/cxx/Debug/4g371p39/obj/armeabi-v7a/libunistyles.so
  : && /opt/android-sdk/ndk/26.1.10909125/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=armv7-none-linux-androideabi24 --sysroot=/opt/android-sdk/ndk/26.1.10909125/toolchains/llvm/prebuilt/linux-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -march=armv7-a -mthumb -Wformat -Werror=format-security  -DFOLLY_NO_CONFIG=1 -DFOLLY_CFG_NO_COROUTINES=1 -DFOLLY_MOBILE=1 -fno-limit-debug-info  -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--no-undefined-version -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments -shared -Wl,-soname,libunistyles.so -o ../../../../build/intermediates/cxx/Debug/4g371p39/obj/armeabi-v7a/libunistyles.so CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/HostStyle.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/MediaQueries.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/StyleSheetRegistry.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/UnistyleWrapper.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/UnistylesCommitHook.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/UnistylesMountHook.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/UnistylesRegistry.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/core/UnistylesState.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridShadowRegistry.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStyleSheet.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/parser/Parser.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/shadowTree/ShadowTreeManager.cpp.o CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o CMakeFiles/unistyles.dir/src/main/cxx/cpp-adapter.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/android/unistylesOnLoad.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridNativePlatformSpec.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesNavigationBarSpec.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesShadowRegistrySpec.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesStatusBarSpec.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesStyleSheetSpec.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesRuntimeSpec.cpp.o CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/android/c++/JHybridNativePlatformSpec.cpp.o  /home/martin/.gradle/caches/8.10.2/transforms/ab3e3d22fe6b83db75ae7058fd428944/transformed/fbjni-0.6.0/prefab/modules/fbjni/libs/android.armeabi-v7a/libfbjni.so  /home/martin/.gradle/caches/8.10.2/transforms/f4e72c61ce8a0261eef3c3938afe876d/transformed/react-android-0.76.3-debug/prefab/modules/jsi/libs/android.armeabi-v7a/libjsi.so  /home/martin/.gradle/caches/8.10.2/transforms/f4e72c61ce8a0261eef3c3938afe876d/transformed/react-android-0.76.3-debug/prefab/modules/reactnative/libs/android.armeabi-v7a/libreactnative.so  -latomic -lm && :
  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::~HybridObject()
  >>> referenced by HybridNavigationBar.h:9 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/../cxx/hybridObjects/HybridNavigationBar.h:9)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(HybridNavigationBar::~HybridNavigationBar())
  >>> referenced by JHybridNativePlatformSpec.hpp:35 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/../nitrogen/generated/android/c++/JHybridNativePlatformSpec.hpp:35)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(margelo::nitro::unistyles::JHybridNativePlatformSpec::~JHybridNativePlatformSpec())
  >>> referenced by HybridStatusBar.h:9 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/../cxx/hybridObjects/HybridStatusBar.h:9)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(HybridStatusBar::~HybridStatusBar())
  >>> referenced 13 more times

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::toString()
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(vtable for HybridStatusBar)
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::unistyles::HybridUnistylesStatusBarSpec-in-HybridStatusBar)
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced 34 more times

  ld.lld: error: undefined symbol: typeinfo for margelo::nitro::HybridObject
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridNavigationBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridNavigationBar)
  >>> referenced 32 more times

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::toObject(facebook::jsi::Runtime&)
  >>> referenced by HybridUnistylesRuntime.cpp:175 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp:175)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(margelo::nitro::unistyles::HybridUnistylesRuntime::createHybridStatusBar(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned int))
  >>> referenced by HybridUnistylesRuntime.cpp:183 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp:183)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(margelo::nitro::unistyles::HybridUnistylesRuntime::createHybridNavigationBar(facebook::jsi::Runtime&, facebook::jsi::Value const&, facebook::jsi::Value const*, unsigned int))

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::loadHybridMethods()
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridNavigationBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridNavigationBar)
  >>> referenced by HybridUnistylesRuntime.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-margelo::nitro::unistyles::HybridUnistylesRuntime)
  >>> referenced 22 more times

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::~HybridObject()
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridNavigationBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridNavigationBar)
  >>> referenced by HybridUnistylesRuntime.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-margelo::nitro::unistyles::HybridUnistylesRuntime)
  >>> referenced 11 more times

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::~HybridObject()
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridNavigationBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridNavigationBar)
  >>> referenced by HybridUnistylesRuntime.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-margelo::nitro::unistyles::HybridUnistylesRuntime)
  >>> referenced 11 more times

  ld.lld: error: undefined symbol: virtual thunk to margelo::nitro::HybridObject::~HybridObject()
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridNavigationBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridNavigationBar)
  >>> referenced by HybridUnistylesRuntime.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-margelo::nitro::unistyles::HybridUnistylesRuntime)
  >>> referenced 11 more times

  ld.lld: error: undefined symbol: virtual thunk to margelo::nitro::HybridObject::~HybridObject()
  >>> referenced by HybridStatusBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStatusBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridStatusBar)
  >>> referenced by HybridNavigationBar.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridNavigationBar.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-HybridNavigationBar)
  >>> referenced by HybridUnistylesRuntime.cpp
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(construction vtable for margelo::nitro::HybridObject-in-margelo::nitro::unistyles::HybridUnistylesRuntime)
  >>> referenced 11 more times

  ld.lld: error: undefined symbol: margelo::nitro::HybridObjectRegistry::hasHybridObject(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&)
  >>> referenced by NativeUnistylesModule.cpp:43 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:43)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)

  ld.lld: error: undefined symbol: margelo::nitro::HybridObjectRegistry::unregisterHybridObjectConstructor(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&)
  >>> referenced by NativeUnistylesModule.cpp:46 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:46)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)
  >>> referenced by NativeUnistylesModule.cpp:47 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:47)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)
  >>> referenced by NativeUnistylesModule.cpp:48 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:48)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)

  ld.lld: error: undefined symbol: margelo::nitro::HybridObjectRegistry::registerHybridObjectConstructor(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, std::__ndk1::function<std::__ndk1::shared_ptr<margelo::nitro::HybridObject> ()>&&)
  >>> referenced by NativeUnistylesModule.cpp:61 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:61)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)
  >>> referenced by NativeUnistylesModule.cpp:64 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:64)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)
  >>> referenced by NativeUnistylesModule.cpp:67 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/src/main/cxx/NativeUnistylesModule.cpp:67)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::UnistylesModule::getBindingsInstaller(facebook::jni::alias_ref<facebook::jni::detail::JTypeFor<facebook::jni::HybridClass<margelo::nitro::unistyles::UnistylesModule, facebook::jni::detail::BaseHybridClass>::JavaPart, facebook::jni::JObject, void>::_javaobject*>)::$_0::operator()(facebook::jsi::Runtime&) const)

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::HybridObject(char const*)
  >>> referenced by HybridUnistylesRuntime.h:17 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/../cxx/hybridObjects/HybridUnistylesRuntime.h:17)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(margelo::nitro::unistyles::HybridUnistylesRuntime::HybridUnistylesRuntime(margelo::nitro::unistyles::JHybridNativePlatformSpec, facebook::jsi::Runtime&, std::__ndk1::function<void (std::__ndk1::function<void (facebook::jsi::Runtime&)>&&)>))
  >>> referenced by HybridStatusBar.h:10 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/../cxx/hybridObjects/HybridStatusBar.h:10)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(HybridStatusBar::HybridStatusBar(margelo::nitro::unistyles::JHybridNativePlatformSpec))
  >>> referenced by HybridStyleSheet.h:20 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/../cxx/hybridObjects/HybridStyleSheet.h:20)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(HybridStyleSheet::HybridStyleSheet(std::__ndk1::shared_ptr<margelo::nitro::unistyles::HybridUnistylesRuntime>, std::__ndk1::shared_ptr<facebook::react::UIManager>))
  >>> referenced 3 more times

  ld.lld: error: undefined symbol: margelo::nitro::HybridObject::getName()
  >>> referenced by HybridFunction.hpp:228 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/HybridFunction.hpp:228)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridUnistylesRuntime.cpp.o:(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> margelo::nitro::HybridFunction::getHybridFuncFullName<margelo::nitro::unistyles::HybridUnistylesRuntime>(margelo::nitro::FunctionKind, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, margelo::nitro::unistyles::HybridUnistylesRuntime*))
  >>> referenced by HybridFunction.hpp:228 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/HybridFunction.hpp:228)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/cxx/hybridObjects/HybridStyleSheet.cpp.o:(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> margelo::nitro::HybridFunction::getHybridFuncFullName<HybridStyleSheet>(margelo::nitro::FunctionKind, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, HybridStyleSheet*))
  >>> referenced by HybridFunction.hpp:228 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/HybridFunction.hpp:228)
  >>>               CMakeFiles/unistyles.dir/src/main/cxx/NativeUnistylesModule.cpp.o:(std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> margelo::nitro::HybridFunction::getHybridFuncFullName<margelo::nitro::unistyles::HybridShadowRegistry>(margelo::nitro::FunctionKind, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&, margelo::nitro::unistyles::HybridShadowRegistry*))
  >>> referenced 5 more times

  ld.lld: error: undefined symbol: margelo::nitro::JSICache::getOrCreateCache(facebook::jsi::Runtime&)
  >>> referenced by JSIConverter+Function.hpp:36 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/JSIConverter+Function.hpp:36)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesStyleSheetSpec.cpp.o:(margelo::nitro::JSIConverter<std::__ndk1::function<void (std::__ndk1::vector<margelo::nitro::unistyles::UnistyleDependency, std::__ndk1::allocator<margelo::nitro::unistyles::UnistyleDependency>> const&)>, void>::fromJSI(facebook::jsi::Runtime&, facebook::jsi::Value const&))
  >>> referenced by JSIConverter+Function.hpp:36 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/JSIConverter+Function.hpp:36)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridNativePlatformSpec.cpp.o:(margelo::nitro::JSIConverter<std::__ndk1::function<void (std::__ndk1::vector<margelo::nitro::unistyles::UnistyleDependency, std::__ndk1::allocator<margelo::nitro::unistyles::UnistyleDependency>> const&, margelo::nitro::unistyles::UnistylesNativeMiniRuntime const&)>, void>::fromJSI(facebook::jsi::Runtime&, facebook::jsi::Value const&))
  >>> referenced by JSIConverter+Function.hpp:36 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/JSIConverter+Function.hpp:36)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridNativePlatformSpec.cpp.o:(margelo::nitro::JSIConverter<std::__ndk1::function<void (margelo::nitro::unistyles::UnistylesNativeMiniRuntime const&)>, void>::fromJSI(facebook::jsi::Runtime&, facebook::jsi::Value const&))

  ld.lld: error: undefined symbol: margelo::nitro::Dispatcher::getRuntimeGlobalDispatcher(facebook::jsi::Runtime&)
  >>> referenced by JSIConverter+Function.hpp:40 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/JSIConverter+Function.hpp:40)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridUnistylesStyleSheetSpec.cpp.o:(margelo::nitro::JSIConverter<std::__ndk1::function<void (std::__ndk1::vector<margelo::nitro::unistyles::UnistyleDependency, std::__ndk1::allocator<margelo::nitro::unistyles::UnistyleDependency>> const&)>, void>::fromJSI(facebook::jsi::Runtime&, facebook::jsi::Value const&))
  >>> referenced by JSIConverter+Function.hpp:40 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/JSIConverter+Function.hpp:40)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridNativePlatformSpec.cpp.o:(margelo::nitro::JSIConverter<std::__ndk1::function<void (std::__ndk1::vector<margelo::nitro::unistyles::UnistyleDependency, std::__ndk1::allocator<margelo::nitro::unistyles::UnistyleDependency>> const&, margelo::nitro::unistyles::UnistylesNativeMiniRuntime const&)>, void>::fromJSI(facebook::jsi::Runtime&, facebook::jsi::Value const&))
  >>> referenced by JSIConverter+Function.hpp:40 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/JSIConverter+Function.hpp:40)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridNativePlatformSpec.cpp.o:(margelo::nitro::JSIConverter<std::__ndk1::function<void (margelo::nitro::unistyles::UnistylesNativeMiniRuntime const&)>, void>::fromJSI(facebook::jsi::Runtime&, facebook::jsi::Value const&))

  ld.lld: error: undefined symbol: margelo::nitro::Logger::nativeLog(margelo::nitro::LogLevel, char const*, std::__ndk1::basic_string<char, std::__ndk1::char_traits<char>, std::__ndk1::allocator<char>> const&)
  >>> referenced by NitroLogger.hpp:35 (/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-nitro-modules/android/build/headers/nitromodules/NitroModules/NitroLogger.hpp:35)
  >>>               CMakeFiles/unistyles.dir/home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/nitrogen/generated/shared/c++/HybridNativePlatformSpec.cpp.o:(void margelo::nitro::Logger::log<>(margelo::nitro::LogLevel, char const*, char const*))
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.

  C++ build system [build] failed while executing:
      /opt/android-sdk/cmake/3.22.1/bin/ninja \
        -C \
        /home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android/.cxx/Debug/4g371p39/armeabi-v7a \
        unistyles
    from /home/martin/workspace/mentorshow/apps/mobile/node_modules/react-native-unistyles/android
```

A similar issue occurs with iOS builds:

```
› Preparing react-native-nitro-modules Pods/NitroModules » NitroModules-Info.plist

❌  (ios/Pods/RCT-Folly/folly/portability/Config.h:20:10)

  18 | 
  19 | #ifndef FOLLY_NO_CONFIG
> 20 | #include <folly/folly-config.h>
     |          ^ 'folly/folly-config.h' file not found
  21 | #endif
  22 | 
  23 | #if __has_include(<features.h>)

❌  (node_modules/react-native-nitro-modules/cpp/core/ArrayBuffer.hpp:11:10)

   9 | 
  10 | #include "OwningReference.hpp"
> 11 | #include <jsi/jsi.h>
     |          ^ could not build module 'jsi'
  12 | #include <thread>
  13 | 
  14 | namespace margelo::nitro {

❌  (ios/Pods/RCT-Folly/folly/portability/Config.h:20:10)

  18 | 
  19 | #ifndef FOLLY_NO_CONFIG
> 20 | #include <folly/folly-config.h>
     |          ^ 'folly/folly-config.h' file not found
  21 | #endif
  22 | 
  23 | #if __has_include(<features.h>)

❌  (node_modules/react-native-nitro-modules/cpp/core/ArrayBuffer.hpp:11:10)

   9 | 
  10 | #include "OwningReference.hpp"
> 11 | #include <jsi/jsi.h>
     |          ^ could not build module 'jsi'
  12 | #include <thread>
  13 | 
  14 | namespace margelo::nitro {

❌  (ios/Pods/RCT-Folly/folly/portability/Config.h:20:10)

  18 | 
  19 | #ifndef FOLLY_NO_CONFIG
> 20 | #include <folly/folly-config.h>
     |          ^ 'folly/folly-config.h' file not found
  21 | #endif
  22 | 
  23 | #if __has_include(<features.h>)

❌  (node_modules/react-native-nitro-modules/cpp/core/ArrayBuffer.hpp:11:10)

   9 | 
  10 | #include "OwningReference.hpp"
> 11 | #include <jsi/jsi.h>
     |          ^ could not build module 'jsi'
  12 | #include <thread>
  13 | 
  14 | namespace margelo::nitro {

❌  (ios/Pods/RCT-Folly/folly/portability/Config.h:20:10)

  18 | 
  19 | #ifndef FOLLY_NO_CONFIG
> 20 | #include <folly/folly-config.h>
     |          ^ 'folly/folly-config.h' file not found
  21 | #endif
  22 | 
  23 | #if __has_include(<features.h>)

❌  (node_modules/react-native-nitro-modules/cpp/core/ArrayBuffer.hpp:11:10)

   9 | 
  10 | #include "OwningReference.hpp"
> 11 | #include <jsi/jsi.h>
     |          ^ could not build module 'jsi'
  12 | #include <thread>
  13 | 
  14 | namespace margelo::nitro {

❌  (ios/Pods/RCT-Folly/folly/portability/Config.h:20:10)

  18 | 
  19 | #ifndef FOLLY_NO_CONFIG
> 20 | #include <folly/folly-config.h>
     |          ^ 'folly/folly-config.h' file not found
  21 | #endif
  22 | 
  23 | #if __has_include(<features.h>)
```

--- a/android/build.gradle
+++ b/android/build.gradle
@@ -12,6 +12,11 @@ buildscript {
     }
 }
 
+def reactNativeArchitectures() {
+    def value = rootProject.getProperties().get("reactNativeArchitectures")
+    return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
+}
+
 apply plugin: 'com.android.library'
 apply plugin: 'org.jetbrains.kotlin.android'
 apply from: '../nitrogen/generated/android/unistyles+autolinking.gradle'
@@ -41,6 +46,7 @@ android {
         externalNativeBuild {
             cmake {
                 arguments "-DANDROID_STL=c++_shared"
+                abiFilters (*reactNativeArchitectures())
             }
         }
     }

